### PR TITLE
EXOMailbox Initial release

### PR DESCRIPTION
**Pull request by levesquesamuel**
#### Pull Request (PR) description
EXOMailbox Initial release

### CODE REVIEW

File: Modules/Microsoft365DSC/DSCResources/MSFT_EXOMailbox/MSFT_EXOMailbox.schema.mof [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    [Write, Description("The DisplayName parameter specifies the display name of the mailbox.")] String DisplayName;
+    [Write, Description("")] String DistinguishedName;
+    [Write, Description("")] String ExchangeVersion;
+    [Write, Description("")] String ExternalDirectoryObjectId;
+    [Write, Description("")] String Guid;
+    [Write, Description("")] String Id;
+    [Write, Description("")] String Name;
+    [Write, Description("")] String OrganizationId;
+    [Write, Description("")] String PrimarySmtpAddress;
+    [Write, Description("")] String RecipientType;
```
Comment (NikCharlebois) [inline Modules/Microsoft365DSC/DSCResources/MSFT_EXOMailbox/MSFT_EXOMailbox.schema.mof:17]: We would need the description for all parameters.

File: Modules/Microsoft365DSC/DSCResources/MSFT_EXOMailbox/MSFT_EXOMailbox.schema.mof [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+{
+    [Key, Description("The Identity parameter specifies the Mailbox that you want to modify.")] String Identity;
+    [Write, Description("Mailbox cannot be created/removed in O365.  This must be set to 'Present'"), ValueMap{"Present"}, Values{"Present"}] String Ensure;
+    [Write, Description("The Alias parameter specifies the Exchange alias (also known as the mail nickname) for the recipient.")] String Alias;
+    [Write, Description("The EmailAddresses parameter specifies all the email addresses (proxy addresses) for the Mailbox")] string EmailAddresses[];
+    [Write, Description("The DisplayName parameter specifies the display name of the mailbox.")] String DisplayName;
+    [Write, Description("")] String DistinguishedName;
+    [Write, Description("")] String ExchangeVersion;
+    [Write, Description("")] String ExternalDirectoryObjectId;
+    [Write, Description("")] String Guid;
```
Comment (NikCharlebois) [inline Modules/Microsoft365DSC/DSCResources/MSFT_EXOMailbox/MSFT_EXOMailbox.schema.mof:12]: If this is only read-only, let's not include it. Also, what is the difference between id and GUID?

File: Modules/Microsoft365DSC/DSCResources/MSFT_EXOMailbox/MSFT_EXOMailbox.schema.mof [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+    [Write, Description("")] String SCLRejectThreshold;
+    [Write, Description("")] String IncludeInGarbageCollection;
+    [Write, Description("")] String IsInactiveMailbox;
+    [Write, Description("")] String IsSoftDeletedByDisable;
+    [Write, Description("")] String IsSoftDeletedByRemove;
+    [Write, Description("")] String WhenSoftDeleted;
+    [Write, Description("")] String ArchiveDatabaseGuid;
+    [Write, Description("")] String DatabaseGuid;
+    [Write, Description("")] String ExchangeGuid;
+
```
Comment (NikCharlebois) [inline Modules/Microsoft365DSC/DSCResources/MSFT_EXOMailbox/MSFT_EXOMailbox.schema.mof:163]: Remove empty line in schema.

File: Modules/Microsoft365DSC/Examples/Resources/EXOMailbox/1-ConfigureMailbox.ps1 [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+        [PSCredential]
+        $credsCredential
+    )
+    Import-DscResource -ModuleName Microsoft365DSC
+
+    node localhost
+    {
+        EXOMailbox "EXOMailbox-Adele Vance"
+        {
+            AcceptMessagesOnlyFrom                 = "";
```
Comment (NikCharlebois) [inline Modules/Microsoft365DSC/Examples/Resources/EXOMailbox/1-ConfigureMailbox.ps1:19]: Let's remove empty string parameters completely

**Review by NikCharlebois — CHANGES_REQUESTED** [review]

**Comment by ykuijs** [comment]
@levesquesamuel Could you please resolve the review comments? After that we can merge your PR.

**Comment by ykuijs** [comment]
And please update your branch with Dev. It currently has conflicts.

**Pull request closed without merging**